app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java

```diff
@@ -1,18 +1,34 @@
 package com.github.houseorganizer.houseorganizer.panels.settings;
 
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
 import android.os.Bundle;
 
 import androidx.annotation.StyleRes;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
 
 import com.github.houseorganizer.houseorganizer.R;
-import com.github.houseorganizer.houseorganizer.util.Util;
+
+import java.util.Locale;
 
 public class ThemedAppCompatActivity extends AppCompatActivity {
+    private SharedPreferences sp;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        sp = PreferenceManager.getDefaultSharedPreferences(this);
         setTheme(getCurrentTheme());
+        Configuration config = getBaseContext().getResources().getConfiguration();
+        Locale locale = new Locale(getCurrentLanguage());
+        Locale.setDefault(locale);
+        config.setLocale(locale);
```

Comment:
Question to understand better, here you get the language from the shared prefs l.24, which returns english as default, but if someone has their phone in italian let's say, does it automatically change to italian and if yes where does it do that or does the person chose in the settings menu ?

Reply:
okay I saw your second sprint task which responds to my question : the user choses in the settings.

Reply:
Yes, we are basically fighting the system. This code is executed in every activity and overrides the default language of the phone